api-catalog-ui/frontend/src/utils/utilFunctions.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+/**
+ * Counts the additional contents
+ * @param service
+ * @returns {{videosCounter: number, useCasesCounter: number, tutorialsCounter: number}}
+ */
+export default function countAdditionalContents(service) {
+    let useCasesCounter = 0;
+    let tutorialsCounter = 0;
+    let videosCounter = 0;
+    if (service) {
+        if ('useCases' in service && service.useCases) {
+            useCasesCounter = service.useCases.length;
+        }
+        if ('tutorials' in service && service.tutorials) {
+            tutorialsCounter = service.tutorials.length;
+        }
+        if ('videos' in service && service.videos) {
+            videosCounter = service.videos.length;
+        }
+    }
+    return { useCasesCounter, tutorialsCounter, videosCounter };
+}
+
+function setMultipleElements(uiConfig) {
+    if (uiConfig.headerColor) {
+        const divider = document.getElementById('separator2');
+        const logoutButton = document.getElementById('go-back-button');
+        const title1 = document.getElementById('title');
+        const swaggerLabel = document.getElementById('swagger-label');
+        const header = document.getElementsByClassName('header');
+        if (header && header.length > 0) {
+            header[0].style.setProperty('background-color', uiConfig.headerColor);
+        }
+        if (divider) {
+            divider.style.setProperty('background-color', uiConfig.headerColor);
+        }
+        if (title1) {
+            title1.style.setProperty('color', uiConfig.headerColor);
+        }
+        if (swaggerLabel) {
+            swaggerLabel.style.setProperty('color', uiConfig.headerColor);
+        }
+        if (logoutButton) {
+            logoutButton.style.setProperty('color', uiConfig.headerColor);
+        }
+    }
+}
+
+/**
+ * Retrieve the logo set in the configuration
+ * @returns {Promise<T>}
+ */
+function fetchImagePath() {
```

Comment:
Is this cached? Or does it re-download the asset every time?

Reply:
Currently, it will re-download it